### PR TITLE
Mct4307 Clear data and condition widget tests

### PR DESCRIPTION
**Pull request by scottbell**
### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [X] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

**Comment by scottbell** [comment]
@nikhilmandlik @unlikelyzero Could you please review this PR? It's the same as:
https://github.com/nasa/openmct/pull/4374
but updated with `master`

File: src/plugins/clearData/pluginSpec.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+            openmct.selection.select(selection);
+
+            openmct.on('start', done);
+            openmct.startHeadless();
+        });
+
+        afterEach(async () => {
+            openmct.router.path = null;
+            await resetApplicationState(openmct);
```
Comment (nikhilmandlik) [inline src/plugins/clearData/pluginSpec.js:80]: I think we can remove `async` `await` and do `return resetApplicationState(openmct);` as it returns promise 😄

**Review by nikhilmandlik — APPROVED** [review]
Small changes/suggestion else LGTM

**Review by unlikelyzero — APPROVED** [review]

**Review by nikhilmandlik — DISMISSED** [review]
LGTM

**Comment by scottbell** [comment]
@nikhilmandlik There were some conflicts caused by the recent PR here:
https://github.com/nasa/openmct/pull/4233
It looks like another test suite was added for condition widgets while this PR was waiting. I've merged the tests together and resolved conflicts. Can you please re-review and if it looks good, merge? Many thanks!

**Review by nikhilmandlik — APPROVED** [review]
LGTM

**Comment by nikhilmandlik** [comment]
### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)

**Pull request merged**